src/Microsoft.Azure.WebJobs.Host/Storage/StorageAccountType.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+namespace Microsoft.Azure.WebJobs.Host.Storage
+{
+    /// <summary>Storage account types.</summary>
+    [Flags]
+    internal enum StorageAccountTypes
```

Comment:
I don't think this enum needs to be a flags. Why can't it be named StorageAccountType, with GeneralPurpose, BlobOnly, Premium? For the checks you need to do, I don't think you need flags.